yt_dlp/extractor/jtbc.py

```diff
@@ -0,0 +1,142 @@
+            'release_date': '20231008',
+            'age_limit': 15,
+            'thumbnail': 'https://fs.jtbc.co.kr//joydata/CP00000001/prog/drama/stronggirlnamsoon/img/20231008_163541_522_1.jpg',
+            'series': '힘쎈여자 강남순',
+        },
+    }, {
+        'url': 'https://vod.jtbc.co.kr/player/program/ep20216733',
+        'md5': '217a6d190f115a75e4bda0ceaa4cd7f4',
+        'info_dict': {
+            'id': 'ep20216733',
+            'ext': 'mp4',
+            'title': '헬로 마이 닥터 친절한 진료실 149회 다시보기',
+            'description': 'md5:1d70788a982dd5de26874a92fcffddb8',
+            'duration': 2720.0,
+            'release_date': '20231009',
+            'age_limit': 15,
+            'thumbnail': 'https://fs.jtbc.co.kr//joydata/CP00000001/prog/culture/hellomydoctor/img/20231009_095002_528_1.jpg',
+            'series': '헬로 마이 닥터 친절한 진료실',
+        },
+    }, {
+        'url': 'https://vod.jtbc.co.kr/player/clip/vo10721270',
+        'md5': '05782e2dc22a9c548aebefe62ae4328a',
+        'info_dict': {
+            'id': 'vo10721270',
+            'ext': 'mp4',
+            'title': '뭉쳐야 찬다3 2회 예고편 - A매치로 향하는 마지막 관문💥',
+            'description': 'md5:d48b51a8655c84843b4ed8d0c39aae68',
+            'duration': 46.0,
+            'release_date': '20231015',
+            'age_limit': 15,
+            'thumbnail': 'https://fs.jtbc.co.kr//joydata/CP00000001/prog/enter/soccer3/img/20231008_210957_775_1.jpg',
+            'series': '뭉쳐야 찬다3',
+        },
+    }, {
+        'url': 'https://tv.jtbc.co.kr/trailer/pr10010392/pm10032526/vo10720912/view',
+        'md5': '217a6d190f115a75e4bda0ceaa4cd7f4',
+        'info_dict': {
+            'id': 'vo10720912',
+            'ext': 'mp4',
+            'title': '아는 형님 404회 예고편 | 10월 14일(토) 저녁 8시 50분 방송!',
+            'description': 'md5:2743bb1079ceb85bb00060f2ad8f0280',
+            'duration': 148.0,
+            'release_date': '20231014',
+            'age_limit': 15,
+            'thumbnail': 'https://fs.jtbc.co.kr//joydata/CP00000001/prog/enter/jtbcbros/img/20231006_230023_802_1.jpg',
+            'series': '아는 형님',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        file_id = self._search_regex(r'data-vod="(VO[0-9]+)"', webpage, 'vod_id')
+
+        metadata = self._download_json(
+            f'https://now-api.jtbc.co.kr/v1/vod/detail?vodFileId={file_id}',
+            video_id, note='Downloading mobile details', fatal=False)
+
+        playback_data = self._download_json(
+            f'https://api.jtbc.co.kr/vod/{file_id}', video_id, note='Downloading VOD playback data')
+
+        subtitles = {}
+        for sub in playback_data.get('tracks', []):
+            subtitles.setdefault(sub.get('label', 'und'), []).append({'url': sub.get('file')})
+
+        formats = []
+        for format_id, stream in traverse_obj(playback_data, ('sources', 'HLS'), default={}).items():
```

Comment:
can do this more safely w/ filtering and transformation in the path
```suggestion
        for format_id, stream in traverse_obj(playback_data, ('sources', 'HLS', lambda _, v: v['file'], {dict.items}, ...)):
```

Reply:
Oh, this does not work as intended.
`traverse_obj(playback_data, ('sources', 'HLS', lambda _, v: v['file'], {dict.items}, ...))` returns a flattened list of tuples:
```
[
  ('file', 'https://...'),
  ('default', False),
  ('streamDefault', False),
  ('label', '360p'),
  ('vodFileId', 'VO10721192'),
  ('file', 'https://...'),
  ('default', False),
  ('streamDefault', True),
  ('label', 720p'),
  ('vodFileId', 'VO10721193'),
]
```

`traverse_obj(playback_data, ('sources', 'HLS', lambda _, v: v['file']))` drops the keys of the initial dict which were used as the format_id and instead returns an array of dicts.

For now, I have adjusted the code to not set the `m3u8_id`. It was superfluous anyway.

Reply:
ah, my mistake

if you want to set `m3u8_id` you can use this
```py
        for format_id, stream in traverse_obj(playback_data, ('sources', 'HLS', {dict.items}, lambda _, v: v[1]['file'])):
```

otherwise if you want to stick with the new simplified way, we can simplify it further
```py
        for stream_url in traverse_obj(media_info, ('movie_list', ..., 'url')):
            location = re.sub(r'/(?:chunk|play)list(?:_pd\d+)?\.m3u8', '/manifest.m3u8', stream_url)
```

Reply:
The simplified code is for MBN :D
I applied the simplification to both.